.github/workflows/run-test-cases.yml

```diff
@@ -58,25 +59,102 @@ on:
       - published
     
 jobs:
+  build-containers:
+    # Build containers only if this is a PR ... otherwise, the containers are pulled ghcr
+    if: >-
+      ( github.event_name == 'pull_request_target' && github.event.pull_request.head.repo.fork == true ) ||
+      ( github.event_name == 'pull_request' && github.event.pull_request.head.repo.fork == false )
+    runs-on: ubuntu-18.04
+    timeout-minutes: 35
+
+    steps:
+    - name: Checkout the merged commit from PR and base branch
+      uses: actions/checkout@v2
+      if: github.event_name == 'pull_request_target'
+      with:
+        # pull_request_target is run in the context of the base repository
+        # of the pull request, so the default ref is master branch and
+        # ref should be manually set to the head of the PR
+        ref: refs/pull/${{ github.event.pull_request.number }}/head
```

Comment:
Whenever checking out the repo, if possible try to set ` persist-credentials: false` like here https://github.com/deislabs/akri/blob/63010918c04ba4042e86953277870e538f0d476e/.github/workflows/build-video-streaming-app-container.yml#L71 for added security. It is enabled by default.

Reply:
👍